Apps/pholetools/oletools_connector.py

```diff
@@ -0,0 +1,229 @@
+# File: oletools_connector.py
+# Copyright (c) 2021 Splunk Inc.
+#
+# Licensed under Apache 2.0 (https://www.apache.org/licenses/LICENSE-2.0.txt)
+
+# Python 3 Compatibility imports
+from __future__ import print_function, unicode_literals
+
+# Phantom App imports
+import phantom.app as phantom
+from phantom.base_connector import BaseConnector
+from phantom.action_result import ActionResult
+from phantom import vault
+
+from oletools_consts import *
+import requests
+import json
+import oletools.oleid
+from oletools.mraptor import MacroRaptor
+from oletools import olevba
+
+
+class RetVal(tuple):
+
+    def __new__(cls, val1, val2=None):
+        return tuple.__new__(RetVal, (val1, val2))
+
+
+class OletoolsConnector(BaseConnector):
+
+    def __init__(self):
+
+        # Call the BaseConnectors init first
+        super(OletoolsConnector, self).__init__()
+
+        self._state = None
+
+    def _get_error_message_from_exception(self, e):
+        """
+        Get appropriate error message from the exception.
+        :param e: Exception object
+        :return: error message
+        """
+        error_code = ERR_CODE_MSG
+        error_msg = ERR_MSG_UNAVAILABLE
+
+        try:
+            if hasattr(e, "args"):
+                if len(e.args) > 1:
+                    error_code = e.args[0]
+                    error_msg = e.args[1]
+                elif len(e.args) == 1:
+                    error_code = ERR_CODE_MSG
+                    error_msg = e.args[0]
+        except:
+            pass
+
+        try:
+            if error_code in ERR_CODE_MSG:
+                error_text = "Error Message: {}".format(error_msg)
+            else:
+                error_text = "Error Code: {}. Error Message: {}".format(error_code, error_msg)
+        except:
+            self.debug_print(PARSE_ERR_MSG)
+            error_text = PARSE_ERR_MSG
+
+        return error_text
+
+    def _handle_test_connectivity(self, param):
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        self.save_progress("Test Connectivity Passed")
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _handle_mraptor_scan(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        summary = action_result.update_summary({})
+
+        vault_id = param['vault_id']
+
+        try:
+            success, message, info = vault.vault_info(vault_id=vault_id, container_id=self.get_container_id(), trace=True)
```

Comment:
Hey @dfederschmidt, I would like to highlight one point regarding this vault file search. Here, it will search for the current container only in Phantom 4.10. So, if you meant to search for all the available containers, it will require some code change as per our best practice for 4.10. Hence, could you please confirm the behavior for this vault file search?

Reply:
In this snippet, the intention is only to search vault artifacts on the current container, not across all containers.

Reply:
cool, thanks for the confirmation.